Framework/Kernel/test/LiveListenerInfoTest.h

```diff
@@ -0,0 +1,177 @@
+    const std::string xml = "<livedata>"
+                            "<connection name='n1' address='a' listener='l' />"
+                            "<connection name='n2' address='A' listener='L' />"
+                            "</livedata>";
+
+    FacilityInfo *fac = nullptr;
+    TS_ASSERT_THROWS_NOTHING(fac = createMinimalFacility(xml));
+
+    InstrumentInfo inst = fac->instruments().front();
+    LiveListenerInfo info;
+
+    info = inst.liveListenerInfo();
+    TS_ASSERT_EQUALS(info.name(), "n1");
+    TS_ASSERT_EQUALS(info.address(), "a");
+    TS_ASSERT_EQUALS(info.listener(), "l");
+
+    info = inst.liveListenerInfo("n1");
+    TS_ASSERT_EQUALS(info.name(), "n1");
+    TS_ASSERT_EQUALS(info.address(), "a");
+    TS_ASSERT_EQUALS(info.listener(), "l");
+
+    info = inst.liveListenerInfo("n2");
+    TS_ASSERT_EQUALS(info.name(), "n2");
+    TS_ASSERT_EQUALS(info.address(), "A");
+    TS_ASSERT_EQUALS(info.listener(), "L");
+
+    TS_ASSERT_THROWS(inst.liveListenerInfo("n3"), std::runtime_error);
+  }
+
+  void test_xml_two_connections_default() {
+    const std::string xml = "<livedata default='n2'>"
+                            "<connection name='n1' address='a' listener='l' />"
+                            "<connection name='n2' address='A' listener='L' />"
+                            "</livedata>";
+
+    FacilityInfo *fac = nullptr;
+    TS_ASSERT_THROWS_NOTHING(fac = createMinimalFacility(xml));
+
+    InstrumentInfo inst = fac->instruments().front();
+    LiveListenerInfo info;
+
+    info = inst.liveListenerInfo();
+    TS_ASSERT_EQUALS(info.name(), "n2");
+    TS_ASSERT_EQUALS(info.address(), "A");
+    TS_ASSERT_EQUALS(info.listener(), "L");
+
+    info = inst.liveListenerInfo("n1");
+    TS_ASSERT_EQUALS(info.name(), "n1");
+    TS_ASSERT_EQUALS(info.address(), "a");
+    TS_ASSERT_EQUALS(info.listener(), "l");
+
+    info = inst.liveListenerInfo("n2");
+    TS_ASSERT_EQUALS(info.name(), "n2");
+    TS_ASSERT_EQUALS(info.address(), "A");
+    TS_ASSERT_EQUALS(info.listener(), "L");
+
+    TS_ASSERT_THROWS(inst.liveListenerInfo("n3"), std::runtime_error);
+  }
+
+  void test_manual_construction() {
+    TS_ASSERT_THROWS_NOTHING(LiveListenerInfo info());
+
+    LiveListenerInfo info;
+    TS_ASSERT_EQUALS(info.name(), "");
+    TS_ASSERT_EQUALS(info.address(), "");
+    TS_ASSERT_EQUALS(info.listener(), "");
+
+    TS_ASSERT_THROWS_NOTHING(info = LiveListenerInfo("l"));
+    TS_ASSERT_EQUALS(info.name(), "");
+    TS_ASSERT_EQUALS(info.address(), "");
+    TS_ASSERT_EQUALS(info.listener(), "l");
+
+    TS_ASSERT_THROWS_NOTHING(info = LiveListenerInfo("l", "a"));
+    TS_ASSERT_EQUALS(info.name(), "");
+    TS_ASSERT_EQUALS(info.address(), "a");
+    TS_ASSERT_EQUALS(info.listener(), "l");
+
+    TS_ASSERT_THROWS_NOTHING(info = LiveListenerInfo("l", "a", "n"));
+    TS_ASSERT_EQUALS(info.name(), "n");
+    TS_ASSERT_EQUALS(info.address(), "a");
+    TS_ASSERT_EQUALS(info.listener(), "l");
+  }
+
+  void test_equality() {
```

Comment:
Overall, good testing. I like the fact you thought to include some logical equality test.